packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -385,14 +418,14 @@ export interface CommonProjectProps {
    *
    * @default - No asset build script.
    */
-  readonly buildScriptAsset?: assets.Asset;
+  readonly buildScript?: assets.ZipDirectoryAsset;
```

Comment:
I realized something: this is basically `codebuild.Source.asset(directory)`. No need for a special prop.

Reply:
What....? I don't understand what this means.... What's the type of the source in the case it's `Source.asset(directory)`?

Reply:
The underlying source type will be S3. Basically the asset will be copied by the CDK to an S3 location and the CodeBuild project will use this location as the source for the build. Magic! 🥇

Reply:
That's a completely different feature though, right? I mean, this is not at all how this feature works now (right now, there is no source in the resulting CodeBuild Project, and the file is downloaded from S3 during build, unzipped, and then the entrypoint is executed).

Does this fall under API cleanup then...? Perhaps I should remove these 2 attributes from Project, and come back to this?

Reply:
Also, what's the relationship between `Source.asset(directory)`, and `buildpsec` in that world?

Reply:
Let’s remove this feature and add it later. You are right.

Reply:
Issue?

Reply:
Removed.

Reply:
Issue: https://github.com/awslabs/aws-cdk/issues/2897